demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -0,0 +1,1062 @@
+            LogError( ( "Error sending command response" ) );
+        }
+    }
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Properties callback handler
+ */
+static AzureIoTResult_t prvProcessProperties( AzureIoTHubClientPropertiesResponse_t * pxMessage,
+                                              AzureIoTHubClientPropertyType_t xPropertyType,
+                                              double * xOutTemperature,
+                                              uint32_t * ulOutVersion )
+{
+    AzureIoTResult_t xResult;
+    AzureIoTJSONReader_t xReader;
+    AzureIoTJSONTokenType_t xTokenType;
+    const uint8_t * pucComponentName = NULL;
+    uint32_t ulComponentNameLength = 0;
+
+    *xOutTemperature = 0.0;
+
+    xResult = AzureIoTJSONReader_Init( &xReader, pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTHubClientProperties_GetPropertiesVersion( &xAzureIoTHubClient, &xReader, pxMessage->xMessageType, ulOutVersion );
+
+    if( xResult != eAzureIoTSuccess )
+    {
+        LogError( ( "Error getting the property version" ) );
+    }
+    else
+    {
+        /* Reset JSON reader to the beginning */
+        xResult = AzureIoTJSONReader_Init( &xReader, pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );
+        configASSERT( xResult == eAzureIoTSuccess );
+
+        while( ( xResult = AzureIoTHubClientProperties_GetNextComponentProperty( &xAzureIoTHubClient, &xReader,
+                                                                                 pxMessage->xMessageType, xPropertyType,
+                                                                                 &pucComponentName, &ulComponentNameLength ) ) == eAzureIoTSuccess )
+        {
+            xResult = AzureIoTJSONReader_TokenType( &xReader, &xTokenType );
+            configASSERT( xResult == eAzureIoTSuccess );
+
+            if( AzureIoTJSONReader_TokenIsTextEqual( &xReader,
+                                                     sampleazureiotPROPERTY_TARGET_TEMPERATURE_TEXT,
+                                                     strlen( sampleazureiotPROPERTY_TARGET_TEMPERATURE_TEXT ) ) )
+            {
+                xResult = AzureIoTJSONReader_NextToken( &xReader );
+                configASSERT( xResult == eAzureIoTSuccess );
+
+                xResult = AzureIoTJSONReader_TokenType( &xReader, &xTokenType );
+                configASSERT( xResult == eAzureIoTSuccess );
+
+                if( xTokenType != eAzureIoTJSONTokenNUMBER )
+                {
+                    LogError( ( "Invalid value for target temperature reached" ) );
+                    xResult = eAzureIoTErrorFailed;
+                    break;
+                }
+
+                /* Get desired temperature */
+                xResult = AzureIoTJSONReader_GetTokenDouble( &xReader, xOutTemperature );
+                configASSERT( xResult == eAzureIoTSuccess );
```

Comment:
how about just using AzureIoTJSONReader_GetTokenDouble and if this fail we print the error that will give us reason why it failed.